### PR TITLE
set up for signing with keyfiles

### PR DESCRIPTION
**Pull request by ceseale**
related to #40 and goes with https://github.com/polyswarm/rust-web3/pull/2

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -89,87 +95,131 @@ fn main() -> Result<(), Error> {
...
+
+    let account = utils::clean_0x(&settings.relay.account)
+        .parse()
+        .or_else(|_| Err(OperationError::InvalidAddress(settings.relay.account.clone())))
+        .unwrap();
+
+    let home_web3 = Web3::new(home_ws.clone());
+    let side_web3 = Web3::new(side_ws.clone());
+
+    home_web3.eth().transaction_count(account, None)
```
Comment (mjkoo) [inline src/main.rs]: This is fine for now but when we merge it let's also file an issue about nonce-resync, judging from polyswarm-client experience that's gonna be an issue

File: src/relay.rs [hunk 2]
```diff
@@ -151,6 +144,10 @@ pub struct Network<T: DuplexTransport> {
     free: bool,
     confirmations: u64,
     anchor_frequency: u64,
+    chain_id: u64,
+    keydir: String,
+    password: String,
+    nonce: Arc<Mutex<U256>>,
```
Comment (mjkoo) [inline src/relay.rs]: Think we could probably get away with an atomic int here, debatable though.

Comment (ceseale) [reply src/relay.rs]: I tried using AtomicU64 but it's experimental and the compiler didn't like it

File: src/main.rs [hunk 3]
```diff
@@ -93,89 +104,137 @@ fn main() -> Result<(), Error> {
     // Set up our two websocket connections on the same event loop
     let mut eloop = tokio_core::reactor::Core::new()?;
     let handle = eloop.handle();
+
     let home_ws = web3::transports::WebSocket::with_event_loop(&settings.relay.homechain.wsuri, &handle)
-        .map_err(SyncFailure::new)?;
+        .map_err(SyncFailure::new)
+        .unwrap();
+
```
Comment (Rizato) [inline src/main.rs]: Why unwrap over the `?`?

Comment (ceseale) [reply src/main.rs]: Oh nice find, I was getting errors requiring that `?` can only be used in a function returning a Result or Option at some point but it's not an issue anymore, I'll throw that back in. :+1:

File: src/relay.rs (hunk 2 above)
Comment (mjkoo) [reply src/relay.rs]: Yeah for some reason the atomic fixed sized ints have been nighly only for forever, AtomicUsize would probably be fine but a bit arch dependant, this also works just is probably less efficient

Comment (ceseale) [reply src/relay.rs]: @mjkoo updated with 7b26efa

**Pull request merged**